src/niscope/system_tests/test_system_niscope.py

```diff
@@ -8,7 +8,54 @@ def session():
         yield simulated_session
 
 
-# Basic usability tests
-def test_load_session(session):
-    pass
+def test_self_test(session):
+    result, message = session.self_test()
+    assert result == 0
+    assert message == 'Scope Self Tests PASSED.'
 
+
+def test_get_channel_name(session):
+    assert session.get_channel_name(2, 3) == '1'
+
+
+def test_reset(session):
+    deault_fetch_relative_to = session.fetch_relative_to
+    assert deault_fetch_relative_to == niscope.FetchRelativeTo.PRETRIGGER
+    session.fetch_relative_to = niscope.FetchRelativeTo.READ_POINTER
+    non_default_acqusition_type = session.fetch_relative_to
+    assert non_default_acqusition_type == niscope.FetchRelativeTo.READ_POINTER
+    session.reset()
+    assert session.fetch_relative_to == niscope.FetchRelativeTo.PRETRIGGER
+
+
+def test_reset_device(session):
+    deault_meas_percentage_method = session.meas_percentage_method
+    assert deault_meas_percentage_method == niscope.PercentageMethod.BASETOP
+    session.meas_percentage_method = niscope.PercentageMethod.MINMAX
+    non_default_meas_percentage_method = session.meas_percentage_method
+    assert non_default_meas_percentage_method == niscope.PercentageMethod.MINMAX
+    session.reset_device()
+    assert session.meas_percentage_method == niscope.PercentageMethod.BASETOP
+
+
+def test_reset_with_defaults(session):
+    deault_meas_time_histogram_high_time = session.meas_time_histogram_high_time
+    assert deault_meas_time_histogram_high_time == 0.0005
+    session.meas_time_histogram_high_time = 0.0010
+    non_default_meas_time_histogram_high_time = session.meas_time_histogram_high_time
+    assert non_default_meas_time_histogram_high_time == 0.0010
+    session.reset_with_defaults()
+    assert session.meas_time_histogram_high_time == 0.0005
+
+
+def test_sample_rate(session):
```

Comment:
FYI Some of these functions will go away per our review of NI-SCOPE, so you can remove / not add the system tests for those. For all the notes yet to be implemented in that branch, let alone in master, see

https://github.com/ni/nimi-python/blob/niscope-addons/NI-SCOPE%20review%20notes.md

Reply:
In order to make this even easier, I have a PR for the removals #604 which I expect to be merged quickly (paging @texasaggie97).